### DISCUSSION TITLE
[Request] Aspect Ratio and Extra pool of words in Basic

### DISCUSSION THREAD

**Discussion by hazmatai7** [AIrjen/OneButtonPrompt#46] [Ideas]
First of all, I love the extension, thanks contributors! 

I wonder if it's possible to set 2 or 3 **aspect ratios to be randomized**: 512 x 512, 512 x 768, 1024 x 512 etc

and also is it possible to ad a box for **"random extra words"**, so that the user throw a few words that will be used at random but not always. I felt the need to add like, "nude" or "2 girls" and it'd work for loras and TIs at random also!

(In the Main tab)

cheers

**Comment by AIrjen** [comment]
Hi hazmatai7,

Both things already exists. For the first one, this is part of the One Button Run and Upscale. Here the aspects ratio's are randomized since this uses the API to call the image processing. This is currently not possible to do in the normal generation process.

For the second request, random extra words, this has already been build and is available to use. But you need to add those as .csv files in the folders. They won't be overwritten during updates. Please check out the documentation I wrote here: https://github.com/AIrjen/OneButtonPrompt/blob/main/user_guides/custom_files.md

- This allows you to extend or replace existing lists.
- There is a specific file already set up for lora's and TI's.
- There are some custom lists that add words into the generation. So you can add your words into one of those as well, and they will be randomly picked up during the prompt generation.

Hope this information helps with whatever you wish to generate.

**Comment by hazmatai7** [comment]
Oh ok... I see it now, thanks.

I just thought of a more intuitive or user/friendly ui adding like those couple input boxes to the Main tab to make it more dynamic or agile.

Since the Main tab is already very intuitive.
